.devcontainer/devcontainer.json

```diff
@@ -5,10 +5,10 @@
 	"build": {
 		"dockerfile": "Dockerfile",
 		"args": {
-			// Update the VARIANT arg to pick a Java version: 8, 11, 14
+			// Update the VARIANT arg to pick a Java version: 8, 11, 17
 			// Append -bullseye or -buster to pin to an OS version.
 			// Use the -bullseye variants on local arm64/Apple Silicon.
-			"VARIANT": "11-bullseye",
+			"VARIANT": "17-bullseye",
```

Comment:
@chrmarti @joshspicer I see in the vscode-dev-containers Java definition, we default to `11-bullseye` too: https://github.com/microsoft/vscode-dev-containers/blob/main/containers/java/.devcontainer/devcontainer.json#L9. Do you think we should update both to 17?

Reply:
All Azure services have now Java 17 support, using the Microsoft build of OpenJDK. It's the current LTS version, and we should now use it by default, instead of an older release.
I believe we should update both, do you want me to do a PR on the other project?

Reply:
Makes sense. Thanks @jdubois! I agree, we should also update vscode-dev-container.

Reply:
Thanks for the insights @jdubois!  Yes, if you put up a PR on vscode-dev-container, we would definitely appreciate that.  

These projects are the source for what we release under mcr.microsoft.com/vscode/devcontainers/java, as well as the devcontainer templates we provide in Codespaces and the VS Code Remote-Containers extension.

@bamurtaugh - how do we sync these `vscode-remote-try-*` repos?  Is it automated or manual?

https://github.com/microsoft/vscode-dev-containers/tree/main/containers/java
https://github.com/microsoft/vscode-dev-containers/tree/main/containers/java-8

Reply:
Awesome! I’ll do the PR in the next few days

Reply:
how do we sync these vscode-remote-try-* repos? Is it automated or manual?

As in how do we sync them into the tooling (i.e. Remote-Containers), or how do we keep them in sync with vscode-dev-containers?

If the latter, it's manual AFAIK. For the former, I think it's automatic, but @chrmarti could confirm.

Reply:
The `vscode-remote-try-*` repositories are only referenced by URL from Remote-Containers, so no update needed on that side.